fgpyo/_requirements.py

```diff
@@ -0,0 +1,24 @@
+"""Enforce requirements."""
+
+from typing import Optional
+
+
+class RequirementError(Exception):
+    """Exception raised when a requirement is not satisfied."""
+
+
+def require(condition: bool, message: Optional[str] = None) -> None:
```

Comment:
Scala has a nice syntax for arguments where the type can be e.g. `message: => str` which means "any expression or function that yields a str".  It's used places like this to defer evaluation of the message to when it's used (i.e. only if the requirement fails).  I'm not sure there's a nice way to do this in python though.

Reply:
`Callable[[], str]` would be the type from `collections.abc.Callable`

Reply:
Added in 6de405e9afb9c76b906fcf4a4c5871ff589acdf4.

Happy to keep or revert if you don't prefer the Python variation of the syntax, up to you!